client/state/with-reducer.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { createElement, Component, PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Higher Order Component to add a reducer to the store's reducer list.
+ * TODO: Add this file to a place accessible by all extensions when it's ready.
+ *
+ * @param { string } reducerName The name for the reducer/key for redux state tree.
+ * @param { func } reducerFunc The function that takes (state, action) and returns state.
+ * @returns { func } A curried function that takes a component to wrap.
+ */
+export default function withReducer( reducerName, reducerFunc ) {
+	return function wrapWithWithReducer( WrappedComponent ) {
+		const wrappedComponentName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
+		const displayName = `WithReducer${ wrappedComponentName }[${ reducerName }]`;
+
+		class WithReducer extends Component {
+			static contextTypes = {
+				store: PropTypes.object,
+			}
+
+			constructor( props, context ) {
+				super( props, context );
+				this.store = props.store || context.store;
+				this.propsMode = Boolean( props.store );
```

Comment:
To my knowledge, no component in Calyspo uses props for this, but it is an accepted Redux pattern. In fact, I patterned this code off of the `react-redux` `connect()` code: https://github.com/reactjs/react-redux/blob/master/src/components/connectAdvanced.js#L122